core/roboconf-dm/src/main/java/net/roboconf/dm/internal/api/impl/InstancesMngrImpl.java

```diff
@@ -401,10 +403,16 @@ private void deployTarget( ManagedApplication ma, Instance scopedInstance ) thro
 		try {
 			// Send the model
 			scopedInstance.setStatus( InstanceStatus.DEPLOYING );
+			String targetId = this.targetsMngr.findTargetId(ma.getApplication(), InstanceHelpers.computeInstancePath( scopedInstance ));
```

Comment:
I suggest you put all of this in a single method, in the target API.
Add one method to get the script resources from a target ID, and another one that gets it for an application and a scoped instance.